lib/providers/lmstudio.ts

```diff
@@ -0,0 +1,154 @@
+import { createOpenAICompatible } from "@ai-sdk/openai-compatible";
+import {
+  multiselect,
+  isCancel,
+  cancel,
+  confirm,
+  text,
+  spinner,
+  note,
+} from "@clack/prompts";
+import type { LanguageModel } from "ai";
+
+export function createLMStudioProvider(
+  baseURL: string = "http://localhost:1234/v1",
+) {
+  return createOpenAICompatible({
+    name: "lmstudio",
+    baseURL,
+  });
+}
+
+export const lmstudio = createLMStudioProvider();
+
+export interface LMStudioModel {
+  id: string;
+  object: string;
+  owned_by: string;
+}
+
+interface LMStudioModelsResponse {
+  object: string;
+  data: LMStudioModel[];
+}
+
+export interface LMStudioConfig {
+  baseURL: string;
+}
+
+export async function fetchLMStudioModels(
+  baseURL: string = "http://localhost:1234/v1",
+): Promise<LMStudioModel[] | null> {
+  try {
+    const response = await fetch(`${baseURL}/models`, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    if (!response.ok) {
+      console.error(
+        `Failed to fetch LM Studio models: ${response.status} ${response.statusText}`,
+      );
+      return null;
+    }
+
+    const data = (await response.json()) as LMStudioModelsResponse;
+    return data.data || [];
+  } catch {
+    return null;
+  }
+}
+
+export async function configureLMStudio(): Promise<LMStudioConfig> {
+  const customUrl = await confirm({
+    message: "Use custom LM Studio URL? (default: http://localhost:1234/v1)",
+    initialValue: false,
+  });
+
+  if (isCancel(customUrl)) {
+    cancel("Operation cancelled.");
+    process.exit(0);
+  }
+
+  let baseURL = "http://localhost:1234/v1";
+
+  if (customUrl) {
+    const urlInput = await text({
+      message: "Enter LM Studio server URL",
+      placeholder: "http://localhost:1234/v1",
+    });
+
+    if (isCancel(urlInput)) {
+      cancel("Operation cancelled.");
+      process.exit(0);
+    }
+
+    baseURL = urlInput || "http://localhost:1234/v1";
+  }
```

Comment:
Instead of using two questions we can ask for the lm studio URL and prefill with the deafault no?